src/main/scala/loamstream/drm/DrmStatus.scala

```diff
@@ -15,30 +15,21 @@ sealed trait DrmStatus {
   import DrmStatus._
 
   def isDone: Boolean = this == Done
-  def isFailed: Boolean = this.isInstanceOf[Failed]
+  def isFailed: Boolean = this == Failed
   def isRequeued: Boolean = this == Requeued
   def isRequeuedHeld: Boolean = this == RequeuedHeld
   def isQueued: Boolean = this == Queued
   def isQueuedHeld: Boolean = this == QueuedHeld
   def isRunning: Boolean = this == Running
-  def isSuspended: Boolean = this.isInstanceOf[Suspended]
-  def isUndetermined: Boolean = this.isInstanceOf[Undetermined]
-  def isDoneUndetermined: Boolean = this.isInstanceOf[DoneUndetermined]
+  def isSuspended: Boolean = this == Suspended
+  def isUndetermined: Boolean = this == Undetermined
+  def isDoneUndetermined: Boolean = this == DoneUndetermined
 
   def notFinished: Boolean = {
     isRequeued || isRequeuedHeld || isQueued || isQueuedHeld || isRunning || isSuspended || isUndetermined
```

Comment:
Not part of the diff, but wondering if this would be better written as `!(isDone || isFailed)` as it's much fewer states to check (if there's a new state in the future, having to remember to add it to this list would be rough. Or - even better - perhaps an overridden method of the trait?

```scala
sealed trait DrmStatus {
  def isFinished: Boolean = false
}

object Done extends DrmStatus {
  override def isFinished: Boolean = true
}
```

Side-note: I've always found that having a boolean `finished` flag for something that can fail gets to be quite ambiguous down the road; if it failed, is that considered "finished" and now I need to check another flag to know if it was successful? Is `Done` the same as `CompletedSuccessfully` or `Finished`? All the states are covered, so it's easy enough to check `isFailed` or `isDone`, but I wonder if `isFinished` might be better named `isFailedOrComplete` or something a bit more unambiguous? Just tossing out ideas.

Reply:
> Not part of the diff, but wondering if this would be better written as !(isDone || isFailed)

That's a good idea.  `notFinished` started out much shorter and grew over several refactoring rounds.